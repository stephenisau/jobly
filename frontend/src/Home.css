.Home {
    height: calc(100vh - 150px);
    display: flex;
    align-items: center;
    min-height: 100vh;
}

#homecard {
    padding: 50px;
    border: 1px solid #FFE5B4;
    padding: 10px;
    box-shadow: 5px 10px 18px black;
    min-height: 250px;
    max-width: 600px;
}
