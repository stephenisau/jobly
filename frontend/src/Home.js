import React, { Component } from "react";
import { Link } from 'react-router-dom';
import UserContext from './UserContext';
import "./Home.css";

class Home extends Component {

  render() {
    return (
      <div className="Home">
        <div className="container text-center card" id="homecard">
          <div className="text-container mt-4">
            <h1 className="mb-4 font-weight-bold">Jobly</h1>
            <p className="lead">All the jobs in one, convenient place.</p>
            <UserContext.Consumer>
              {currentUser => (
                <React.Fragment>
                  {currentUser
                    ? <h2>Welcome Back {currentUser.user.first_name}!</h2>
                    : <Link className="btn btn-primary mx-auto" id="home-login" to="/login">Log in</Link>
                  }
                </React.Fragment>
              )}
            </UserContext.Consumer>
          </div>
        </div>
      </div>
    );
  }
}

export default Home;