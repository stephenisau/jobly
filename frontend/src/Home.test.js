import React from "react";
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Home from './Home';

it('renders without crashing', function() {
  shallow(<Home />);
})

it('matches snapshot', function() {
  let wrapper = shallow(<Home />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
})