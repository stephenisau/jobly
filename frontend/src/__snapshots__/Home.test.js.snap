// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Home"
>
  <div
    className="container text-center card"
    id="homecard"
  >
    <div
      className="text-container mt-4"
    >
      <h1
        className="mb-4 font-weight-bold"
      >
        Jobly
      </h1>
      <p
        className="lead"
      >
        All the jobs in one, convenient place.
      </p>
      <ContextConsumer>
        <Component />
      </ContextConsumer>
    </div>
  </div>
</div>
`;
